sklearn/pipeline.py

```diff
@@ -209,7 +200,7 @@ def _fit(self, X, y=None, **fit_params):
             if transformer is None:
                 pass
             else:
-                if memory.cachedir is None:
+                if getattr(memory, 'cachedir', True) is None:
```

Comment:
The default value `True` is a bit magical/misleading here ... maybe one of the (rare) cases where `hasattr` is slightly easier to read:

```py
if hasattr(memory, 'cachedir') and memory.cachedir is None:
```